src/radical/pilot/agent/resource_manager/pbspro.py

```diff
@@ -20,35 +19,35 @@ class PBSPro(ResourceManager):
     #
     def _init_from_scratch(self, rm_info: RMInfo) -> RMInfo:
 
-        # TODO: $NCPUS?!?! = 1 on archer
+        try:
+            vnodes, rm_info.cores_per_node = self._parse_pbspro_vnodes()
+            nodes = [(node, rm_info.cores_per_node) for node in vnodes]
 
-        cpn = os.environ.get('NUM_PPN') or os.environ.get('SAGA_PPN')
-        if not cpn:
-            raise RuntimeError('$NUM_PPN and $SAGA_PPN not set!')
+        except (IndexError, ValueError) as exc:
+            self._log.debug_2('exec_vnodes not detected')
```

Comment:
What does `exec_vnodes` mean?  Maybe `cannot parse qstat output` or something like that?

Reply:
with this I actually want to point to `exec_vnodes` (PBSPro specific attribute), since that is what we look for and `qstat` is just a tool, thus we reach this message not only if `qstat` failed, but also if `exec_vnodes` is failed to be parsed

Reply:
I actually don't know what `exec_vnodes` is.  from looking at the code I assume that's some qstat output field - but that may not be obvious to the user.

Reply:
btw, that is under `DEBUG_2`, kinda just for thorough debug, it is not needed for regular (`info`) and debug messages - extra info to let us know why it was switched from `_parse_pbspro_vnodes` to using `PBS_NODEFILE`

Reply:
Fair point...